packages/device_info_plus/device_info_plus_windows/pubspec.yaml

```diff
@@ -13,7 +13,7 @@ dependencies:
   ffi: ^2.0.1
   flutter:
     sdk: flutter
-  win32: ^2.7.0
+  win32: ^3.0.0
```

Comment:
Please don't set artificial range restrictions on widely-used packages. See [this comment](https://github.com/ThexXTURBOXx/flutter_web_auth_2/commit/3a7b846677f1fd4326a7a641607a1ba8941dca18#r82923160) and [this comment](https://github.com/flutter/plugins/pull/6352#discussion_r961905860)

Reply:
It's not an artificial range restriction in my opinion. Most people will try to use the newest version of all packages. This just updates the `win32` dependency. But yes, I will change it to the same version range as before

Reply:
> It's not an artificial range restriction in my opinion.

How is preventing this package from being used in conjunction with any package that uses `win32` 2.x when this package doesn't require 3.x not an artificial restriction?

Reply:
Let's view it the other way around: If some package updates to the newest version (which is what you should primarily support in the first place), this package would restrict the way forward. As already said, I will loosen the version range

Reply:
Reverted change

Reply:
I think you are misunderstanding my comment; I'm not saying that using ^3 instead of ^2 is an artificial restriction, I'm saying that using ^3 instead of ">=2 <4" is an artificial restriction (because this package is completely compatible with both 2.x and 3.x).

> If some package updates to the newest version (which is what you should primarily support in the first place)

The reason I'm commenting on these PRs is that using a range instead of an unnecessary minimum directly supports that goal. https://github.com/ThexXTURBOXx/flutter_web_auth_2/issues/6 is a case of someone who was *prevented* from using the latest version of a package (in that case, yours) because it used ^3 instead of a range when a range would have worked equally well. Replicating that unnecessary restriction into more packages would make more people have conflicts like that and/or be on older version of more packages for longer.